### PR TITLE
upgrade default TiDB version to v3.0.5

### PR DESCRIPTION
**Pull request by shonge**
### What problem does this PR solve? 
https://github.com/pingcap/tidb-operator/issues/1089
### What is changed and how does it work?
Upgrade TiDB component's images.

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - No code

Code changes

 - Has Helm charts change
 - Has Go code change
 - Has documents change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note

 ```

### CODE REVIEW

**Comment by shonge** [comment]
@weekface PTAL

File: CHANGELOG.md [hunk 1]
```diff
@@ -257,7 +257,7 @@ The `monitor` section in new `values.yaml`:
 monitor:
   create: true
    initializer:
-     image: pingcap/tidb-monitor-initializer:v3.0.1
+     image: pingcap/tidb-monitor-initializer:v3.0.5
```
Comment (weekface) [inline CHANGELOG.md]: This is a changelog content, no need to modify.

**Pull request merged**